api/pkg/service/catalog/catalog.go

```diff
@@ -106,6 +107,50 @@ func (s *service) RefreshAll(ctx context.Context, p *catalog.RefreshAllPayload)
 	return res, nil
 }
 
+// List the all type of errors of a catalog
+func (s *service) CatalogError(ctx context.Context, p *catalog.CatalogErrorPayload) (*catalog.CatalogErrorResult, error) {
+
+	log := s.Logger(ctx)
+	db := s.DB(ctx)
+
+	ctg := model.Catalog{}
+	if err := db.Where(&model.Catalog{Name: p.CatalogName}).Model(&model.Catalog{}).First(&ctg).Error; err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return nil, catalogNotFoundErr(p.CatalogName)
+		}
+		log.Error(err)
+		return nil, internalError
+	}
+
+	var allErrors []string
+
+	db.Model(&model.CatalogError{CatalogID: ctg.ID}).Distinct("type").Find(&allErrors)
```

Comment:
what if we fetch by only one query and then segregate?
`select * from CatalogError  where catalog_id=<>`
Create 3 arrays for each type ..loop on catalogError array and add error in respective array
and if array is not empty -> add in response object?
wdyt?